pkg/upgrade/upgrader/jiva_upgrade.go

```diff
@@ -630,6 +646,148 @@ func jivaUpgrade(pvName, openebsNamespace string, utaskObj *utask.UpgradeTask) (
 		return options.utaskObj, err
 	}
 
+	err = options.cleanup(openebsNamespace)
+	if err != nil {
+		return options.utaskObj, err
+	}
+
 	klog.Info("Upgrade Successful for", pvName)
 	return options.utaskObj, nil
 }
+
+func (j *jivaVolumeOptions) cleanup(openebsNamespace string) error {
+	var err error
+	if j.replicaObj.version == currentVersion {
+		err = deployClient.WithNamespace(j.ns).Delete(j.replicaObj.name, &metav1.DeleteOptions{})
+		if err != nil {
+			return err
+		}
+	}
+	if j.controllerObj.version == currentVersion && j.ns != openebsNamespace {
+		err = deployClient.WithNamespace(j.ns).Delete(j.controllerObj.name, &metav1.DeleteOptions{})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (j *jivaVolumeOptions) preMigration(openebsNamespace string) error {
+	var err error
+	if j.replicaObj.version == currentVersion {
+		err = scaleDeploy(j.controllerObj.name, j.ns, ctrlDeployLabel, 0)
+		if err != nil {
+			return err
+		}
+	}
+	if j.controllerObj.version == currentVersion {
+		err = scaleDeploy(j.replicaObj.name, j.ns, replicaDeployLabel, 0)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (j *jivaVolumeOptions) migrateReplica(openebsNamespace string) error {
+	if j.replicaObj.version == currentVersion {
+		klog.Infof("splitting replica deployment")
+		deployObj, err := deployClient.WithNamespace(j.ns).Get(j.replicaObj.name)
+		if err != nil {
+			return err
+		}
+		nodeNames := deployObj.Spec.Template.Spec.Affinity.NodeAffinity.
+			RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms[0].MatchExpressions[0].Values
+		var one int32 = 1
+		for i, node := range nodeNames {
+			replicaDeploy := deployObj.DeepCopy()
+			replicaDeploy.Name = replicaDeploy.Name + "-" + strconv.Itoa(i+1)
+			replicaDeploy.Namespace = openebsNamespace
+			replicaDeploy.ResourceVersion = ""
+			replicaDeploy.Spec.Replicas = &one
```

Comment:
as part of postMigration i.e., after replicaPatch is done, scale up the deployments